### PR TITLE
As ast implementation

### PR DESCRIPTION
**Pull request by pareeohnos**
This provides an initial implementation of the AST rendered as initially discussed in #145 
I feel however this should be rewritten again, but this is a starting point as a proof of concept. It's not 100% working and there are a couple of failing tests still

### CODE REVIEW

File: lib/earmark.ex [hunk 1]
```diff
@@ -318,7 +318,8 @@ defmodule Earmark do
 
   defp _as_html(lines, options) do
     start_link()
-    {blocks, context} = parse(lines, options)
+    ast = {blocks, context} = parse(lines, options)
```
Comment (RobertDober) [inline lib/earmark.ex]: I think for the time being we should leave the working code alone.

**Comment by RobertDober** [comment]
Hmm that is not what I was thinking about, for safety and clearity reasons I wanted an `Earmark.as_ast(...)` which would not touch any existing code.

Do you think your work can be adapted to that approach?

File: lib/earmark.ex (hunk 1 above)
Comment (pareeohnos) [reply lib/earmark.ex]: Ah this was just me testing. I've got a bunch of `IO.inspect` lines throughout the code testing things out, this was just one of them :)

**Comment by pareeohnos** [comment]
Closing in favour of a rewrite

**Pull request closed without merging**